python/packages/autogen-ext/src/autogen_ext/models/azure/_azure_ai_client.py

```diff
@@ -0,0 +1,483 @@
+
+        # Copy the create args and overwrite anything in extra_create_args
+        create_args = self._create_args.copy()
+        create_args.update(extra_create_args)
+
+        if self.capabilities["vision"] is False:
+            for message in messages:
+                if isinstance(message, UserMessage):
+                    if isinstance(message.content, list) and any(isinstance(x, Image) for x in message.content):
+                        raise ValueError("Model does not support vision and image was provided")
+
+        if json_output is not None:
+            if self.capabilities["json_output"] is False and json_output is True:
+                raise ValueError("Model does not support JSON output")
+
+            if json_output is True and "response_format" not in create_args:
+                create_args["response_format"] = ChatCompletionsResponseFormatJSON()
+
+        if self.capabilities["json_output"] is False and json_output is True:
+            raise ValueError("Model does not support JSON output")
+        if self.capabilities["function_calling"] is False and len(tools) > 0:
+            raise ValueError("Model does not support function calling")
+
+        azure_messages_nested = [to_azure_message(msg) for msg in messages]
+        azure_messages = [item for sublist in azure_messages_nested for item in sublist]
+
+        task: Task[ChatCompletions]
+
+        if len(tools) > 0:
+            converted_tools = convert_tools(tools)
+            task = asyncio.create_task(
+                self._client.complete(messages=azure_messages, tools=converted_tools, **create_args)
+            )
+        else:
+            task = asyncio.create_task(
+                self._client.complete(
+                    messages=azure_messages,
+                    **create_args,
+                )
+            )
+
+        if cancellation_token is not None:
+            cancellation_token.link_future(task)
+
+        result: ChatCompletions = await task
+
+        usage = RequestUsage(
+            prompt_tokens=result.usage.prompt_tokens if result.usage else 0,
+            completion_tokens=result.usage.completion_tokens if result.usage else 0,
+        )
+
+        choice = result.choices[0]
+        if choice.finish_reason == CompletionsFinishReason.TOOL_CALLS:
+            assert choice.message.tool_calls is not None
+
+            content = [
+                FunctionCall(
+                    id=x.id,
+                    arguments=x.function.arguments,
+                    name=normalize_name(x.function.name),
+                )
+                for x in choice.message.tool_calls
+            ]
+            finish_reason = "function_calls"
+        else:
+            finish_reason = choice.finish_reason.value
+            content = choice.message.content or ""
+
+        response = CreateResult(
+            finish_reason=finish_reason,  # type: ignore
```

Comment:
Let's use the latest update to canonicalize the finish_reason. See OpenAIChatCompletionClient